Sources/XCBuildSupport/XcodeBuildSystem.swift

```diff
@@ -133,20 +131,28 @@ public final class XcodeBuildSystem: SPMBuildCore.BuildSystem {
             supportedArchitectures: [],
```

Comment:
While you're in this area, can targetArchitecture also be updated to the host architecture (x86_64 or arm64) instead of hardcoded, if it's readily available?

Reply:
Absolutely — there's a whole range of fixes here, but I wanted to parcel them out across a couple of different PRs.  This particular one would unblock some CI problems, as alluded to, but I plan to put up another PR that cleans up some of the other hardcoded things here.  Also `-toolchain` and `-sdk` are not respected, for example.

Reply:
Put up as a separate PR here: https://github.com/apple/swift-package-manager/pull/3596